tests/src/test/kotlin/dev/restate/e2e/CancelInvocationTest.kt

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2023 - Restate Software, Inc., Restate GmbH
+//
+// This file is part of the Restate e2e tests,
+// which are released under the MIT license.
+//
+// You can find a copy of the license in file LICENSE in the root
+// directory of this repository or package, or at
+// https://github.com/restatedev/e2e/blob/main/LICENSE
+
+package dev.restate.e2e
+
+import com.google.protobuf.Empty
+import dev.restate.admin.api.InvocationApi
+import dev.restate.admin.client.ApiClient
+import dev.restate.admin.model.TerminationMode
+import dev.restate.e2e.services.awakeableholder.AwakeableHolderServiceGrpc
+import dev.restate.e2e.services.awakeableholder.hasAwakeableRequest
+import dev.restate.e2e.services.awakeableholder.unlockRequest
+import dev.restate.e2e.services.canceltest.*
+import dev.restate.e2e.services.canceltest.CancelTestProto.BlockingOperation
+import dev.restate.e2e.utils.*
+import dev.restate.generated.IngressGrpc.IngressBlockingStub
+import dev.restate.generated.invokeRequest
+import io.grpc.Channel
+import java.net.URL
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.TimeoutCancellationException
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import org.awaitility.kotlin.await
+import org.awaitility.kotlin.matches
+import org.awaitility.kotlin.untilCallTo
+import org.junit.jupiter.api.extension.RegisterExtension
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+
+class JavaCancelInvocationTest : BaseCancelInvocationTest() {
+  companion object {
+    @RegisterExtension
+    val deployerExt: RestateDeployerExtension =
+        RestateDeployerExtension(
+            RestateDeployer.Builder()
+                .withServiceEndpoint(
+                    Containers.javaServicesContainer(
+                        "java-cancel-invocation",
+                        CancelTestServiceGrpc.SERVICE_NAME,
+                        BlockingServiceGrpc.SERVICE_NAME))
+                .withServiceEndpoint(
+                    Containers.nodeServicesContainer(
+                        "awakeable-holder", AwakeableHolderServiceGrpc.SERVICE_NAME))
+                .build())
+  }
+}
+
+class NodeCancelInvocationTest : BaseCancelInvocationTest() {
+  companion object {
+    @RegisterExtension
+    val deployerExt: RestateDeployerExtension =
+        RestateDeployerExtension(
+            RestateDeployer.Builder()
+                .withServiceEndpoint(
+                    Containers.nodeServicesContainer(
+                        "node-cancel-invocation",
+                        CancelTestServiceGrpc.SERVICE_NAME,
+                        BlockingServiceGrpc.SERVICE_NAME,
+                        AwakeableHolderServiceGrpc.SERVICE_NAME))
+                .build())
+  }
+}
+
+abstract class BaseCancelInvocationTest {
+  @ParameterizedTest(name = "cancel blocked invocation on {0}")
+  @MethodSource("blockingOperations")
```

Comment:
You can use `@EnumSource`

Reply:
There are too many annotations in the Java ecosystem. Will update it.